src/net_processing.cpp

```diff
@@ -94,144 +94,156 @@ void static ProcessGetData(CNode *pfrom, const Consensus::Params &consensusParam
-                // never disconnect whitelisted nodes
-                static const int nOneWeek = 7 * 24 * 60 * 60; // assume > 1 week = historical
-                if (fSend && CNode::OutboundTargetReached(true) &&
-                    (((pindexBestHeader != nullptr) &&
-                         (pindexBestHeader.load()->GetBlockTime() - mi->GetBlockTime() > nOneWeek)) ||
-                        inv.type == MSG_FILTERED_BLOCK) &&
-                    !pfrom->fWhitelisted)
-                {
-                    LOG(NET, "historical block serving limit reached, disconnect peer %s\n", pfrom->GetLogName());
+                    // disconnect node in case we have reached the outbound limit for serving historical blocks
+                    // never disconnect whitelisted nodes
+                    static const int nOneWeek = 7 * 24 * 60 * 60; // assume > 1 week = historical
+                    if (fSend && CNode::OutboundTargetReached(true) &&
+                        (((pindexBestHeader != nullptr) &&
+                             (pindexBestHeader.load()->GetBlockTime() - mi->GetBlockTime() > nOneWeek)) ||
+                            inv.type == MSG_FILTERED_BLOCK) &&
+                        !pfrom->fWhitelisted)
+                    {
+                        LOG(NET, "historical block serving limit reached, disconnect peer %s\n", pfrom->GetLogName());
 
-                    // disconnect node
-                    pfrom->fDisconnect = true;
-                    fSend = false;
-                }
-                // Avoid leaking prune-height by never sending blocks below the
-                // NODE_NETWORK_LIMITED threshold.
-                // Add two blocks buffer extension for possible races
-                if (fSend && !pfrom->fWhitelisted &&
-                    ((((nLocalServices & NODE_NETWORK_LIMITED) == NODE_NETWORK_LIMITED) &&
-                        ((nLocalServices & NODE_NETWORK) != NODE_NETWORK) &&
-                        (chainActive.Tip()->nHeight - mi->nHeight > (int)NODE_NETWORK_LIMITED_MIN_BLOCKS + 2))))
-                {
-                    LOG(NET, "Ignore block request below NODE_NETWORK_LIMITED threshold from peer=%d\n",
-                        pfrom->GetId());
-                    // disconnect node and prevent it from stalling (would
-                    // otherwise wait for the missing block)
-                    pfrom->fDisconnect = true;
-                    fSend = false;
-                }
-                // Pruned nodes may have deleted the block, so check whether
-                // it's available before trying to send.
-                if (fSend && (mi->nStatus & BLOCK_HAVE_DATA))
-                {
-                    // Send block from disk
-                    CBlock block;
-                    if (!ReadBlockFromDisk(block, mi, consensusParams))
+                        // disconnect node
+                        pfrom->fDisconnect = true;
```

Comment:
Why disconnect if we end up not having the block?  The problem is on our side

Reply:
I think you might be thinking of something else...this is old code that's been there and disconnects the peer if we've reached the block serving limit that the node operator has configured..I dont' think you see it used much but it was something put in by core to prevent to much IBD peers from using up your outbound bandwidth.

Also this is for processing getdata requests, so we're sending here, not receiving.